### PR TITLE
Add Salesforce target docs

### PR DESCRIPTION
**Pull request by odacremolbap**
@daceynolan maybe we can use this PR to improve docs step by step based on your analysis doc. I have found what is missing in that doc but no proposal, should we try to tackle some of the points you mentione for this PR?

### CODE REVIEW

**Comment by daceynolan** [comment]
@odacremolbap I would love this to be a group effort. I only gave suggestions on how it can be improved. As far as how we want to standardize the docs, that has not yet been decided. We should set a meeting time to have everyone explain what they would like to see happen in the docs. I can create a time and ask what works for everyone. I will look at this and we can always make changes later.

**Comment by odacremolbap** [comment]
> @odacremolbap I would love this to be a group effort. I only gave suggestions on how it can be improved. As far as how we want to standardize the docs, that has not yet been decided. We should set a meeting time to have everyone explain what they would like to see happen in the docs. I can create a time and ask what works for everyone. I will look at this and we can always make changes later.

Sounds good.
I'll ask for a "standard review" of this one then.

File: docs/targets/salesforce.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    - Select the Connected App from the list and at the click on `Manage`.
+    - Click `Manage Profiles`.
+    - Add permissions on the data this user will have access to.
+    - Save.
+
+3. Retrieve OAuth data to configure Triggermesh Target.
+
+   - Select the Connected App from the list and at the click on `View`.
+   - Copy `Consumer Key`
+   - Reveal and copy `Consumer Secret`
```
Comment (daceynolan) [inline docs/targets/salesforce.md:44]: Same for at the top, this is showing as `5 Reveal and copy Consumer Secret`

**Review by daceynolan — CHANGES_REQUESTED** [review]
It looks like number is off

File: docs/targets/salesforce.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+Salesforce target uses [OAuth JWT credentials][salesforce-oauth-jwt] for service authentication.
+
+1. First, you will need to generate an X509 certificate for signing and verifying requests.
+We will be using `OpenSSL` but any other certificate generation tool should work.
+
+```sh
+openssl req -x509 -sha256 -nodes -days 36500 -newkey rsa:2048 -keyout tm-sf.key -out tm-sf.crt
+```
+
+2. At Salesforce site select `Setup > Apps > App Manager`, click on `New Connected App`.
```
[the remark this answers is not in the document]
Comment (odacremolbap) [reply docs/targets/salesforce.md]: https://github.com/triggermesh/docs/pull/107/commits/101eb49184fa2455621b7d04200c5c1deadde984

File: docs/targets/salesforce.md (hunk 1 above)
Comment (odacremolbap) [reply docs/targets/salesforce.md]: https://github.com/triggermesh/docs/pull/107/commits/101eb49184fa2455621b7d04200c5c1deadde984

**Review by sameersbn — APPROVED** [review]
lgtm

**Pull request merged**